ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/SessionProxy.java

```diff
@@ -54,11 +56,21 @@
 
     private final TransactionHelper transactionHelper;
 
+    /**
+     * @param xaSession XA session that needs to be proxied.
+     * @param transactionHelper utility to make transaction resources registration easier.
+     */
     public SessionProxy(XASession xaSession, TransactionHelper transactionHelper) {
         this.xaSession = xaSession;
         this.transactionHelper = transactionHelper;
     }
 
+    /**
+     * Simply close proxied session if there is no active transaction. Or if transaction exists, delist session's XA resource
+     * and register {@link SessionClosingSynchronization}.
```

Comment:
how about explicitly saying what the registration is for, ie:
"and register a {@link SessionClosingSynchronization} to close the proxied session."